commands/http/multifilereader.go

```diff
@@ -64,13 +64,23 @@ func (mfr *MultiFileReader) Read(buf []byte) (written int, err error) {
 
 		// handle starting a new file part
 		if !mfr.closed {
-			if file.IsDirectory() {
+
+			var contentType string
+			if s, ok := file.(*files.Symlink); ok {
+				mfr.currentFile = s
+
+				// TODO(why): this is a hack. pick a real contentType
+				contentType = "symlink"
```

Comment:
+1 for symlink as a regular file with ModeSymlink.
Though I think it is possible to write a more lightweight version of that PR (and part of this PR) without the fileinfo.go

+1 vote for sending an archive to the api pipe (tar for now, later dag).
This will decouple the current api writing/porting into api rest parsing & porting the dag archive extractor library (tar for now).